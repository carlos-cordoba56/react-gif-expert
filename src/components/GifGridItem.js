import React from 'react'
import propTypes from 'prop-types';

const GifGridItem = ({title, url}) => {
    // console.log({id, title, url})


    return (
        <div className='card animate__bounce'>
            <img src={ url } alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: propTypes.element.isRequired,
    url: propTypes.element.isRequired
}

export default GifGridItem;