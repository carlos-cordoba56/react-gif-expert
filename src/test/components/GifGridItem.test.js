import React from 'react'
import GifGridItem from '../../components/GifGridItem'
import { shallow } from 'enzyme'

describe('Prueba en <GifGridItem />', () => {

    const title = 'Un titulo'
    const url = 'http://localhost/algo.jpg'
    const wrapper = shallow( <GifGridItem title={title} url={url} /> )

    test('Debe de mostrar el componente correctamente', () => {
        
        
        expect( wrapper ).toMatchSnapshot();

    })

    test('Debe de tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p')
        expect( p.text().trim() ).toBe( 'title' )
    })

    test('Debe coincidir el url y title de la imagen', () => {
        const img = wrapper.find('img');
        // console.log( img.prop('src') )
        expect( img.prop('src') ).toBe(url)
        expect( img.prop('alt') ).toBe(title)
    })

    test('Debe coincidir la clase', () => {
        const div = wrapper.find('div');

        expect(div.prop('className')).toBe('card')
    })
    
    
    
    
})
