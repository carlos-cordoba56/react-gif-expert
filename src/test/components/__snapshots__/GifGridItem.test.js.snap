// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__bounce"
>
  <img />
  <p />
</div>
`;
